src/mongroup.cpp

```diff
@@ -235,6 +244,24 @@ bool MonsterGroup::IsMonsterInGroup( const mtype_id &mtypeid ) const
     return false;
 }
 
+int MonsterGroup::event_adjusted_freq_total( holiday event ) const
+{
+    std::string opt = get_option<std::string>( "EVENT_SPAWNS" );
+    if( opt != "monsters" && opt != "both" ) {
+        event = holiday::none;
```

Comment:
We should at least have minimal performance impact when event spawns are disabled.
```suggestion
        return freq_total;
```

Reply:
@kevingranade Cheers! As you suggested, I added a map to cache freq totals. The function now returns early when possible, and uses the cached value for the event otherwise. I haven't noticed too much difference between the performance of this branch versus master, but hopefully this makes the impact as minimal as possible.